src/libraries/Common/src/Interop/Unix/System.Native/Interop.Stat.cs

```diff
@@ -53,13 +54,13 @@ internal enum FileStatusFlags
             HasBirthTime = 1,
         }
 
-        [DllImport(Libraries.SystemNative, EntryPoint = "SystemNative_FStat", SetLastError = true)]
-        internal static extern int FStat(SafeHandle fd, out FileStatus output);
+        [GeneratedDllImport(Libraries.SystemNative, EntryPoint = "SystemNative_FStat", SetLastError = true)]
+        internal static partial int FStat(SafeHandle fd, out FileStatus output);
 
-        [DllImport(Libraries.SystemNative, EntryPoint = "SystemNative_Stat", SetLastError = true)]
-        internal static extern int Stat(string path, out FileStatus output);
+        [GeneratedDllImport(Libraries.SystemNative, EntryPoint = "SystemNative_Stat", CharSet = CharSet.Ansi, SetLastError = true)]
```

Comment:
Why is `CharSet = CharSet.Ansi` needed here?

Reply:
We had the p/invoke source generators avoid making any assumptions about `CharSet` if it isn't set, so it requires explicitly specifying either `CharSet` or `MarshalAs` when marshalling `string`/`char`.